go/parser/parser.go

```diff
@@ -216,6 +217,14 @@ func (p *parser) declare1(decl ast.Node, scope *ast.Scope, kind ast.ObjKind, ide
 }
 
 func (p *parser) redeclared(ident *ast.Ident, prev *ast.Object, reason string) {
+	if f := p.fset.File(ident.Pos()); isPlatformDependent(f) {
+		last := prev
```

Comment:
or, since we don't care about ordering:

    ident.Next = prev.Next
    prev.Next = ident